iothub/device/src/ClientOptions.cs

```diff
@@ -21,5 +23,11 @@ public class ClientOptions
         /// in the transport settings set for the non-file upload operations.
         /// </summary>
         public Http1TransportSettings FileUploadTransportSettings { get; set; } = new Http1TransportSettings();
+
+        /// <summary>
+        /// The default configuration of MessageId for every message sent by the device or module client instance.
+        /// The default behavior is that MessageId is settable only by the user, by calling <see cref="Message.MessageId"/>.
+        /// </summary>
+        public MessageIdDefault MessageIdDefault { get; set; } = MessageIdDefault.Null;
```

Comment:
Suggestions for renaming `MessageIdDefault`?

Reply:
```csharp
public enum MessageIdDefaultValue
{
    Unset,
    AssignedRandomGuidBySdk,
};
```